### DISCUSSION TITLE
Anyway to hide the building?

### DISCUSSION THREAD

**Discussion by Dorbmon** [scripthookvdotnet/scripthookvdotnet#1293] [Q&A] [answered]
Actually, We can get buildings by World.getAllBuildings(). However, is there any way to hide them?

**Comment by kagikn** [comment, marked as answer]
Actually, I have no idea how you can make buildings invisible or delete them. I tried removing the first bit of [CEntity + 0x2C] as `CPhysical` instances (vehicles, peds, and props/objects) won't get rendered without that flag (`IS_ENTITY_VISIBLE` checks that flag too), but no observable changes on `CBuilding`s.

You might not be able to find anything useful until I research `phInstGta` (`CEntity` contains a pointer to a `phInstGta` at [CEntity + 0x30]) or if you analyze the `CEntity` struct via `MemoryAddress` property on your own... `CBuilding` takes only 0xD0 bytes per instance and a instance of it doesn't contain much info compared to `CPhysical`.
To tell the truth, I added `Building`, `AnimatedBuilding`, and `InteriorInstance` classes since Zolika's Trainer has wallhacks for them and I thought they might be useful for developers with reverse engineer skill (I yoinked memory signature stuff from that trainer).

**Reply by Dorbmon** [reply]
Thank you!
